fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PlanTranslatorContext.java

```diff
@@ -287,16 +287,19 @@ public SlotDescriptor createSlotDesc(TupleDescriptor tupleDesc, SlotReference sl
             slotDescriptor.setLabel(slotReference.getName());
         } else {
             slotRef = new SlotRef(slotDescriptor);
-            if (slotReference.hasSubColPath()) {
-                slotDescriptor.setSubColLables(slotReference.getSubColPath());
-                slotDescriptor.setMaterializedColumnName(slotRef.getColumnName()
-                            + "." + String.join(".", slotReference.getSubColPath()));
-            }
         }
         slotRef.setTable(table);
         slotRef.setLabel(slotReference.getName());
         this.addExprIdSlotRefPair(slotReference.getExprId(), slotRef);
-        slotDescriptor.setIsNullable(slotReference.nullable());
+        if (slotReference.hasSubColPath()) {
+            slotDescriptor.setSubColLables(slotReference.getSubColPath());
+            slotDescriptor.setMaterializedColumnName(slotRef.getColumnName()
+                        + "." + String.join(".", slotReference.getSubColPath()));
+            // always nullable at present
+            slotDescriptor.setIsNullable(true);
```

Comment:
if variant slot always nullable, u should set it correctly in Nereids, because nereids need nullable info todo some optimization. if u set uncorrect nullable info, we will get wrong plan sometimes

Reply:
fixed